chips/nrf52/src/uart.rs

```diff
@@ -121,6 +120,10 @@ impl UART {
         let regs = unsafe { &*self.regs };
         let tx = regs.event_endtx.get() != 0;
 
+        if regs.event_error.get() == 1 {
+            panic!("uart error event\r\n");
```

Comment:
I have checked this now there exist four different error modes:
- Overrun
- Parity
- Framing
- Break

As I have understood it these are not possible in TX-mode and in the current code I haven't enabled "ERROR events" at all thus it is not possible. Thus, the check is completely useless 🥇 

Should remove the error check completely? Or remove I enable error events and keep the error checking?

BTW, [#511](https://github.com/helena-project/tock/pull/511) seem to break the uart code on nRF52. Time for debugging yikes 🔨